src/Controller/FeedbackController.php

```diff
@@ -0,0 +1,438 @@
+     *     )
+     * )
+     * @SWG\Response(
+     *     response=404,
+     *     description="Not found",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="code", type="integer", example=404),
+     *     @SWG\Property(property="message", type="string", example="Not found!"),
+     *     )
+     * )
+     * @SWG\Response(
+     *     response="400",
+     *     description="Bad Request"
+     * )
+     * @SWG\Response(
+     *     response="200",
+     *     description="Successfull operation!",
+     *     @SWG\Schema(
+     *     @SWG\Property(property="message", type="string", example="Feedback submitted successfully!"),
+     *     )
+     * )
+     * @param Activity $activity
+     * @param User $userTo
+     * @param Request $request
+     * @param ValidationErrorSerializer $validationErrorSerializer
+     * @return JsonResponse
+     * @throws ORMException
+     * @throws OptimisticLockException
+     */
+    public function addFeedback(
+        Activity $activity,
+        User $userTo,
+        Request $request,
+        ValidationErrorSerializer $validationErrorSerializer
+    ): JsonResponse {
+        $authenticatedUser = $this->getUser();
+
+        $rights = $this->accessRightsPolicy->canGiveFeedback($activity, $authenticatedUser, $userTo);
+
+        if ($rights === false) {
+            return new JsonResponse([
+                'code' => Response::HTTP_PRECONDITION_FAILED,
+                'message' => 'You were not involved or the job is not yet finished!'
+            ], Response::HTTP_PRECONDITION_FAILED);
+        }
+
+        $data = $request->getContent();
+
+        /** @var DeserializationContext $context */
+        $context = DeserializationContext::create()->setGroups(array('AddFeedback'));
+
+        $feedbackDTO = $this->serializer->deserialize(
+            $data,
+            FeedbackDTO::class,
+            'json',
+            $context
+        );
+
+        $errors = $this->validator->validate($feedbackDTO, null, ['AddFeedback']);
+
+        if (count($errors) > 0) {
+            return new JsonResponse(
+                [
+                    'code' => Response::HTTP_BAD_REQUEST,
+                    'message' => 'Bad Request',
+                    'errors' => $validationErrorSerializer->serialize($errors)
+                ],
+                Response::HTTP_BAD_REQUEST
+            );
+        }
+
+        $newFeedback = $this->feedbackTransformer->addFeedback($feedbackDTO, $authenticatedUser, $activity, $userTo);
+        $this->feedbackRepository->save($newFeedback);
+
+        if ($userTo->getStars() === 0.0) {
+            $userTo->setStars($newFeedback->getStars());
+        } else {
+            $countFeedback = $this->feedbackRepository->countFeedback($userTo)->getQuery()->getSingleScalarResult();
+            $newStars = (($userTo->getStars() * ($countFeedback - 1)) + $newFeedback->getStars()) / $countFeedback;
```

Comment:
1. getQuery() and getSingleScalarResult() should always be in a repository.
2. What's up with this hard math logic? You will slowly over time get a wrong value because of estimation. The correct way is to have FeedbackRepository calculate this for you via a query. Something like `SELECT sum(stars)/count(id) as stars from feedback where user_id = ?`
3. If 2 will be implemented you won't have this `if-else`

Don't forget about the editFeedback()